File: benchmarks/python/12_bigint_overflow.py
def main():
    value = 1 << 100
    checksum = 0
    i = 0
    while i < 200000:
        value = value + 123456789
        value = value - 98765432
        checksum += value % 97
        i += 1
    print(checksum)


if __name__ == "__main__":
    main()

File: benchmarks/c/12_bigint_overflow.c
// BigInt Overflow Path Benchmark - C Implementation

#include <stdint.h>
#include <stdio.h>

int main() {
    __int128 value = ((__int128)1 << 100);
    long long checksum = 0;

    for (int64_t i = 0; i < 200000; i++) {
        value = value + 123456789;
        value = value - 98765432;
        checksum += (long long)(value % 97);
    }

    printf("bigint overflow checksum: %lld\n", checksum);
    return 0;
}
